compat/src/react-18-hooks.js

```diff
@@ -0,0 +1,33 @@
+import { useState, useLayoutEffect, useEffect, useCallback } from 'preact/hooks';
+
+export function useDeferredValue(val) {
+	return val;
+}
+
+export function useTransition() {
+  return [false, useCallback(cb => { cb(); }, [])];
+}
+
+export const useInsertionEffect = useEffect;
+
+export function useSyncExternalStore(subscribe, getSnapshot) {
+	const [state, setState] = useState(getSnapshot);
+	const value = getSnapshot();
+
+	useLayoutEffect(() => {
+		if (value !== state) {
+			setState(value);
+		}
+	}, [value]);
+
+	useEffect(() => {
+		return subscribe(() => {
+			setState(getSnapshot());
+		});
+	}, [subscribe, getSnapshot]);
+
+	return state;
+}
+
+// TODO: useId... This could be a bit more challenging as RTS does not have
```

Comment:
Can we re-use vnodeId here? For example (pseudo code):
```js
export const useId = () => '' + vnodeId;
```

Reply:
Not sure if that would work - every time JSX is reevaluated it generates new VNodes with new IDs, and the IDs are monotonic with no reset options.